<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.13.2" xml:lang="en-US">
  <compounddef id="d2/d82/classsatox_1_1database_1_1DatabaseOperationException" kind="class" language="C++" prot="public">
    <compoundname>satox::database::DatabaseOperationException</compoundname>
    <basecompoundref refid="d4/d35/classsatox_1_1database_1_1DatabaseException" prot="public" virt="non-virtual">satox::database::DatabaseException</basecompoundref>
    <includes refid="d9/dba/database_2include_2satox_2database_2error_8hpp" local="no">error.hpp</includes>
    <sectiondef kind="public-func">
      <memberdef kind="function" id="d2/d82/classsatox_1_1database_1_1DatabaseOperationException_1a5867e7ad0f3abdb3802519c6df78ec6a" prot="public" static="no" const="no" explicit="no" inline="yes" virt="non-virtual">
        <type></type>
        <definition>satox::database::DatabaseOperationException::DatabaseOperationException</definition>
        <argsstring>(const std::string &amp;message=&quot;&quot;)</argsstring>
        <name>DatabaseOperationException</name>
        <qualifiedname>satox::database::DatabaseOperationException::DatabaseOperationException</qualifiedname>
        <param>
          <type>const std::string &amp;</type>
          <declname>message</declname>
          <defval>&quot;&quot;</defval>
        </param>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="src/database/include/satox/database/error.hpp" line="64" column="5" bodyfile="src/database/include/satox/database/error.hpp" bodystart="64" bodyend="65"/>
        <references refid="d4/d35/classsatox_1_1database_1_1DatabaseException_1ad07f1221f1ac70dcf918e0dc3ac009ee" compoundref="d9/dba/database_2include_2satox_2database_2error_8hpp" startline="35" endline="36">satox::database::DatabaseException::DatabaseException</references>
        <references refid="d4/d35/classsatox_1_1database_1_1DatabaseException_1a88813c28bec819804de644f1a11b49a9" compoundref="d9/dba/database_2include_2satox_2database_2error_8hpp" startline="46" endline="48">satox::database::DatabaseException::message</references>
        <references refid="db/dfd/namespacesatox_1_1database_1a39a52ecf35dfba5b1e247d97a3c9c198a8f8d49d3d0842358e0fe8f43601d8537" compoundref="d9/dba/database_2include_2satox_2database_2error_8hpp" startline="23">satox::database::OPERATION_FAILED</references>
      </memberdef>
    </sectiondef>
    <briefdescription>
    </briefdescription>
    <detaileddescription>
    </detaileddescription>
    <inheritancegraph>
      <node id="2">
        <label>satox::database::DatabaseException</label>
        <link refid="d4/d35/classsatox_1_1database_1_1DatabaseException"/>
        <childnode refid="3" relation="public-inheritance">
        </childnode>
      </node>
      <node id="1">
        <label>satox::database::DatabaseOperationException</label>
        <link refid="d2/d82/classsatox_1_1database_1_1DatabaseOperationException"/>
        <childnode refid="2" relation="public-inheritance">
        </childnode>
      </node>
      <node id="3">
        <label>std::exception</label>
      </node>
    </inheritancegraph>
    <collaborationgraph>
      <node id="2">
        <label>satox::database::DatabaseException</label>
        <link refid="d4/d35/classsatox_1_1database_1_1DatabaseException"/>
        <childnode refid="3" relation="public-inheritance">
        </childnode>
      </node>
      <node id="1">
        <label>satox::database::DatabaseOperationException</label>
        <link refid="d2/d82/classsatox_1_1database_1_1DatabaseOperationException"/>
        <childnode refid="2" relation="public-inheritance">
        </childnode>
      </node>
      <node id="3">
        <label>std::exception</label>
      </node>
    </collaborationgraph>
    <location file="src/database/include/satox/database/error.hpp" line="62" column="1" bodyfile="src/database/include/satox/database/error.hpp" bodystart="62" bodyend="66"/>
    <listofallmembers>
      <member refid="d4/d35/classsatox_1_1database_1_1DatabaseException_1ac4b03e2d9ce1c676a728d3d06475bdc8" prot="public" virt="non-virtual"><scope>satox::database::DatabaseOperationException</scope><name>code</name></member>
      <member refid="d4/d35/classsatox_1_1database_1_1DatabaseException_1ad07f1221f1ac70dcf918e0dc3ac009ee" prot="public" virt="non-virtual"><scope>satox::database::DatabaseOperationException</scope><name>DatabaseException</name></member>
      <member refid="d2/d82/classsatox_1_1database_1_1DatabaseOperationException_1a5867e7ad0f3abdb3802519c6df78ec6a" prot="public" virt="non-virtual"><scope>satox::database::DatabaseOperationException</scope><name>DatabaseOperationException</name></member>
      <member refid="d4/d35/classsatox_1_1database_1_1DatabaseException_1a88813c28bec819804de644f1a11b49a9" prot="public" virt="non-virtual"><scope>satox::database::DatabaseOperationException</scope><name>message</name></member>
      <member refid="d4/d35/classsatox_1_1database_1_1DatabaseException_1a86a99a4206d19745d7cda6e4bc9190a5" prot="public" virt="non-virtual"><scope>satox::database::DatabaseOperationException</scope><name>what</name></member>
    </listofallmembers>
  </compounddef>
</doxygen>
